src/dctkit/dec/flat.py

```diff
@@ -2,10 +2,12 @@
 from dctkit.dec import cochain as C
 from jax import Array, vmap
 from functools import partial
+from typing import Callable, Dict, Optional
 
 
-def flat(c: C.CochainP0 | C.CochainD0, weights: Array,
-         edges: C.CochainP1V | C.CochainD1V) -> C.CochainP1 | C.CochainD1:
+def flat(c: C.CochainP0 | C.CochainD0, weights: Array, edges: C.CochainP1V |
+         C.CochainD1V, weighted_I: Optional[Callable] = None,
```

Comment:
weighted_I sounds like "weighted_Identity": the name is misleading. Why not calling it "interp_fun"? What is the meaning of "weighted" here?

Reply:
Done